amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java

```diff
@@ -239,39 +278,47 @@ private void executeAction(final List<NodeRef> actionNodes)
         RetryingTransactionCallback<Boolean> processTranCB = () -> {
             for (NodeRef actionNode : actionNodes)
             {
-                if (nodeService.exists(actionNode))
+                if (!nodeService.exists(actionNode))
+                {
+                    continue;
+                }
+
+                final String dispAction = (String) nodeService.getProperty(actionNode, PROP_DISPOSITION_ACTION);
+
+                // Run disposition action
+                if (dispAction == null || !dispositionActions.contains(dispAction))
```

Comment:
could we have the situation when the ```dispAction!=null && !dispositionAction.contains(dispAction)``` what is happening now comparing with what was previously ```dispAction != null && dispositionActions.contains(dispAction)```

Reply:
As discussed, this situation is taken into account.